src/file.cpp

```diff
@@ -336,14 +337,30 @@ namespace libtorrent
 		return int(size);
 	}
 
-#ifdef TORRENT_WINDOWS
-	std::string convert_separators(std::string p)
+	template <typename T>
+	std::string convert_from_native_path(T* s);
+
+	template <>
+	std::string convert_from_native_path(char const* s) { return convert_from_native(s); }
+
+	template <>
+	std::string convert_from_native_path(char* s) { return convert_from_native(s); }
+
+#if defined TORRENT_WINDOWS && TORRENT_USE_WSTRING
+	template <>
+	std::string convert_from_native_path(wchar_t const* s) { return convert_from_wstring(s); }
+
+	template <>
+	std::string convert_from_native_path(wchar_t* s) { return convert_from_wstring(s); }
```

Comment:
again, I think the non-const overload (or, what should be an overload) is suspect. As far as I can tell, it's redundant and mostly dangerous